Introduction to FINOS CCC (#242)
ScottLogic/blog: _posts/2024-09-23-intro-finos-ccc.md

In summary, FINOS CCC project aims to establish a set of guidelines that enforce security, compliance, and governance for public cloud services used by financial institutions.

## Why do we need it?

Financial service sector retains sensitive data about their customers. They store personal identification information of the customer such as full name, date of birth, social security numbers, national identification numbers, home address, email addresses, phone numbers, etc. They also sometimes store customer income and credit data such as employment status, employer details, salary information, other income sources, credit score, defaults and bankruptcies, etc. Not only that, but financial services also usually have records of customer in-store & online purchase history, withdrawals, deposits, investment activities and related transactional and behavioral patterns. In a nutshell, they hold a vast amount of sensitive customer information, which could cause severe reputation and financial damage in an unfortunate event of a data breach.

> In the past, regulatory requirements for financial institutes mandated that data be stored in highly secured on-premise data centres. However, with the growing adoption of cloud services, the financial services sector is increasingly moving towards the public cloud. Key benefits driving this shift include agility, scalability, cost optimization, accelerated innovation, geographic availability, and enhanced resilience. But the controls and services offered. Some of the potential drawbacks and challenges, particularly given the sensitive nature of the financial data are security concerns, compliance and regulatory challenges, loss of control, downtime, data privacy risks, vendor lock-in and skills gap. Let’s elaborate more on few key concerns in moving financial data to the public cloud. Despite the robust security measures cloud providers offer, financial data is highly sensitive, and breaches can be devastating. Public cloud environments are multi-tenant, meaning data from multiple organizations shares the same infrastructure. While cloud providers enforce strict isolation, the risk of data leakage still exists. The cloud provider’s employees may have access to critical data, posing potential insider threats. When it comes to regulatory challenges, regulations in some regions require that sensitive financial data must be stored within national borders. Cloud providers may not offer appropriate data centre locations, or ensuring compliance with data residency laws can be complex. Auditing and proving compliance in cloud environments can be more challenging, especially with limited visibility into the provider’s operations.

Reviewer: `But the controls and services offered.` - not sure I follow this bit?

> Jim Adams, CTO and Head of Technology Infrastructure at Citi, the world’s fifth largest bank, said, “There is a need for a Cloud Standard that will improve certain security and control measures across the Financial Services industry, whilst simplifying and democratizing access for all institutions to operate and benefit by leveraging the public cloud. It is important to collaborate with our peers to ensure consistency across cloud service providers, ensuring the industry can realize true multi-cloud strategies,” in his statement on July 27, 2023 .

Reviewer: use `>` for quotes, e.g.
> "A thing was said"

Also a link / footnote might be good for this one if you have it

FINOS CCC project uses [OSCAL](https://pages.nist.gov/OSCAL/) (Open Security Controls Assessment Language) developed by NIST (National Institute of Standards and Technology) as their control language. OSCAL utilizes a machine-readable format for defining controls, which facilitates automated assessments, reporting, automated generation of compliance documentation and many more.

The project also aims to validate controls through a series of tests. If you are aware of the controls required in your public cloud, you can use the tests provided in the CCC standard to verify whether those controls are properly implemented.

It is envisaged that eventually, CCC will offer certification for CSPs who conform to the standard.

## Controls, Threats and Features

> To provide you with a clearer understanding of what a control is, let's take a closer look at a specific example: the control that prevent bucket deletion through irrevocable bucket retention policy in object storage. This control is designed to ensure that critical storage resources, like data buckets, are not accidentally or maliciously deleted, thereby protecting the integrity and availability of stored data. By examining this control, we can better understand what are the associated threats and how to mitigate those risks in cloud environments.

Reviewer: the control that prevent**s** bucket...

> This control defined in the `controls.yaml` under object storage is mapped to a specific threat within the system, identified as `CCC.TH06`, which we will explore in more detail later. Additionally, this control is mapped to a NIST control, specifically identified as `PR.DS-1`, which is part of the framework's guidelines for protecting data. There are also specific methods to test whether this control is effectively implemented within your cloud service provider, ensuring that it meets security and compliance standards.

Reviewer: This control, defined in a file named `controls.yaml` under object storage, is mapped to a specific threat within the system...

~~~ yaml
id: CCC.ObjStor.C03
title: Prevent bucket deletion through irrevocable bucket retention policy
control_family: Data
objective: |
    Ensure that object storage bucket is not deleted after creation,
    and that the preventative measure cannot be unset.
threats:
    - CCC.TH06
nist_csf: PR.DS-1
test_requirements:
    tlp_green:
    01: |
        Verify that the bucket cannot be deleted as long as the retention policy is in effect.
    02: |
        Confirm that the bucket retention policy cannot be modified or unset.
~~~

> Let’s examine the threat `CCC.TH06` in  `common-threats.yaml` file, which pertains to the accidental or malicious deletion of data. This threat highlights the potential risk where important data could be lost due to human error or intentional actions by bad actors. This particular threat is also linked to the MITRE ATT&CK framework under the ID `T1485`, which refers to data destruction as a method used by attackers to disrupt operations or erase traces of their activities. Additionally, this threat is mapped to specific feature within the system `CCC.ObjStor.F11`. 

Reviewer: "...mapped to specific feature within the system" - what is the system in this context?

Might be just me but I find putting the example first followed by the description a bit more easy to follow, otherwise you're talking about something I haven't yet seen

According to the Linux Foundation [announcement](https://www.linuxfoundation.org/press/finos-announces-open-standards-project-for-financial-services-common-cloud-controls) on July 27, 2023 Jim Adams, CTO and Head of Technology Infrastructure at Citi, the world’s fifth largest bank, stated

> “There is a need for a Cloud Standard that will improve certain security and control measures across the Financial Services industry, whilst simplifying and democratizing access for all institutions to operate and benefit by leveraging the public cloud. It is important to collaborate with our peers to ensure consistency across cloud service providers, ensuring the industry can realize true multi-cloud strategies,”

## Goals of FINOS CCC

Base on the Common Cloud Control GitHub repo the project aims to fulfil following goals, 

* Defining Best Practices Around Cloud Security
* One Target For CSPs To Conform To
* Sharing The Burden Of A Common Definition
* A Path Towards Common Implementation
* A Path Towards Certification

> or a more detailed explanation of each project goal and how they are achieved, please refer to the [README.md](https://github.com/finos/common-cloud-controls) file in the project's GitHub repository.

Reviewer: minor one but the GitHub repo main page displays the README by default so no need to call out the README explicitly I don't think. It also might be worth making this a footnote as it's referenced again further down

> ## Timeline and Contributors

Reviewer: Answer might be no but, are there any other key dates to make this a bit more of a timeline?

Author: I don't think so :(
First release is yet to come.

## Controls, Threats and Features

To provide you with a clearer understanding of what a control is, let's take a closer look at a specific example: the control that prevents bucket deletion through irrevocable bucket retention policy in object storage. This control is designed to ensure that critical storage resources, like data buckets, are not accidentally or maliciously deleted, thereby protecting the integrity and availability of stored data. By examining this control, we can better understand what are the associated threats and how to mitigate those risks in cloud environments.

~~~ yaml
id: CCC.ObjStor.C03
title: Prevent bucket deletion through irrevocable bucket retention policy
control_family: Data
objective: |
    Ensure that object storage bucket is not deleted after creation,
    and that the preventative measure cannot be unset.
threats:
    - CCC.TH06
nist_csf: PR.DS-1
test_requirements:
    tlp_green:
    01: |
        Verify that the bucket cannot be deleted as long as the retention policy is in effect.
    02: |
        Confirm that the bucket retention policy cannot be modified or unset.
~~~

> This control defined in the file named `controls.yaml` under object storage, is mapped to a specific threat within the standard, identified as `CCC.TH06`, which we will explore in more detail later. Additionally, this control is mapped to a NIST control, specifically identified as `PR.DS-1`, which is part of the framework's guidelines for protecting data. There are also specific methods to test whether this control is effectively implemented within your cloud service provider, ensuring that it meets security and compliance standards.

Reviewer: might be worth linking the NIST control to make it a bit clearer? https://csf.tools/reference/nist-cybersecurity-framework/v1-1/pr/pr-ds/pr-ds-1/

For more details refer to the project's [GitHub](https://github.com/finos/common-cloud-controls) page.

## Scott Logic FINOS CCC Team

Scott Logic was one of the first organizations to collaborate with the FINOS Foundation in establishing the FINOS CCC, under the leadership of Colin Eberhardt and Robert Griffiths. Robert, serving as the project’s sponsorship lead within Scott Logic, played a pivotal role in guiding this significant initiative as a key member of the FINOS CCC Steering Committee. Alongside Rob, Stevie Shiells, who chairs the Community Structure working group, and I, as the chair of the Taxonomy working group, represent Scott Logic in driving this open-source project. We've also received numerous invaluable contributions from our team at Scott Logic over time, including Joshua Isted, Cara Fisher, David Ogle, Mike Smith, Euthyme Ziogas, Daniel Moorhouse, and Ivan Mladjenovic. If you're interested in supporting Scott Logic's vision of empowering open source projects, reach out to any of us to get started on contributing to this exciting initiative.

> ---

Reviewer: might be just me but it feels like the post is missing a closing paragraph to wrap it up. The last sentence here is good to end on but maybe could use some final thoughts on the project in a nutshell and how you've found working on it for example?

Author: I added a Join us section after this to looks like a closing para. Can you take a look to see whether it serves the purpose.

Reviewer: Yeah that's good - I guess I was thinking, you open the post from a very personal perspective before diving into the project itself. It might be nice to go back to that at the end and put a line or two in reflecting on your personal experience now you have been involved for a while. It's fine as it is though to be clear - just a suggestion 😄

---
title: Introducing FINOS Common Cloud Controls (CCC)
categories: 
- Open Source
author: smendis-scottlogic
summary: FINOS Common Cloud Controls (CCC) is an open standard by FINOS, to describe consistent controls for compliance public cloud deployments in financial services sector. The project is sponsored by Scott Logic, aligning with its mission to promote and support open-source initiatives. This is an effort to introduce FINOS CCC and its goals.

> ---

Author: I thought we have promised 2 full time resources to the project. That's sponsored work, right?

> ---

Reviewer 2: Ha ha - yes, I suppose that's very true! You convinced me :-)

This statement outlines several important aspects of the project. First and foremost, it is designed to cater specific needs of the **financial services sector**, which includes banking, insurance, investment and wealth management, mortgage lending, and more. An **open standard** refers to a set of guidelines or specifications developed collaboratively that can be used freely or with minimal restrictions. **Consistent controls** imply standardized security, compliance, and governance measures applied uniformly across the infrastructure, applications and processes. These controls ensure that policies related to data protection, access management, auditing, encryption, and monitoring are reliably implemented. **Compliance public cloud deployments** involve using public cloud services, such as AWS, Microsoft Azure, or Google Cloud, that meet the regulatory and legal standards required by regulating authorities. 

In summary, FINOS CCC project aims to establish a set of guidelines that enforce security, compliance, and governance for public cloud services used by financial institutions.

## Why do we need it?

> Financial service sector retains sensitive data about their customers. They store personal identification information of the customer such as full name, date of birth, social security numbers, national identification numbers, home address, email addresses, phone numbers, etc. They also sometimes store customer income and credit data such as employment status, employer details, salary information, other income sources, credit score, defaults and bankruptcies, etc. Not only that, but financial services also usually have records of customer in-store & online purchase history, withdrawals, deposits, investment activities and related transactional and behavioral patterns. In a nutshell, they hold a vast amount of sensitive customer information, which could cause severe reputation and financial damage in an unfortunate event of a data breach.

Reviewer 3: ```suggestion
The financial services sector retains sensitive data about their customers. They store personally identifiable information of the customer such as name, date of birth, social security numbers, national identification numbers, home address, email addresses, phone numbers, etc. They also sometimes store customer income and credit data such as employment status, employer details, salary information, other income sources, credit score, defaults and bankruptcies, etc. Not only that, but financial services also usually have records of customer in-store & online purchase history, withdrawals, deposits, investment activities and related transactional and behavioral patterns. In a nutshell, they hold a vast amount of sensitive customer information, which could cause severe reputational and financial damage in an unfortunate event of a data breach.
```

In the past, regulatory requirements for financial institutes mandated that data be stored in highly secured on-premise data centres. However, with the growing adoption of cloud services, the financial services sector is increasingly moving towards the public cloud. Key benefits driving this shift include agility, scalability, cost optimization, accelerated innovation, geographic availability, and enhanced resilience. Some of the potential drawbacks and challenges, particularly given the sensitive nature of the financial data are security concerns, compliance and regulatory challenges, loss of control, downtime, data privacy risks, vendor lock-in and skills gap. 

![PROS_AND_CONS_OF_CLOUD]({{ site.github.url }}/smendis-scottlogic/assets/pros-n-cons.png)

> Let’s elaborate more on few key concerns in moving financial data to the public cloud. Despite the robust security measures cloud providers offer, financial data is highly sensitive, and breaches can be devastating. Public cloud environments are multi-tenant, meaning data from multiple organizations shares the same infrastructure. While cloud providers enforce strict isolation, the risk of data leakage still exists. The cloud provider’s employees may have access to critical data, posing potential insider threats. When it comes to regulatory challenges, regulations in some regions require that sensitive financial data must be stored within national borders. Cloud providers may not offer appropriate data centre locations, or ensuring compliance with data residency laws can be complex. Auditing and proving compliance in cloud environments can be more challenging, especially with limited visibility into the provider’s operations.

Reviewer 3: ```suggestion
Let’s elaborate more on few key concerns in moving financial data to the public cloud. Despite the robust security measures cloud providers offer, financial data is highly sensitive, and breaches can be devastating. Public cloud environments are multi-tenant, meaning data from multiple organizations shares the same infrastructure. While cloud providers enforce strict isolation, the risk of data leakage still exists. The cloud provider’s employees may have access to critical data, posing potential insider threats. When it comes to regulatory challenges, regulations in some regions require that sensitive financial data must be stored within national borders. Cloud providers may not offer appropriate data centre locations, or ensure compliance with data residency laws can be complex. Auditing and proving compliance in cloud environments can be more challenging, especially with limited visibility into the provider’s operations.
```

According to the Linux Foundation [announcement](https://www.linuxfoundation.org/press/finos-announces-open-standards-project-for-financial-services-common-cloud-controls) on July 27, 2023 Jim Adams, CTO and Head of Technology Infrastructure at Citi, the world’s fifth largest bank, stated

> “There is a need for a Cloud Standard that will improve certain security and control measures across the Financial Services industry, whilst simplifying and democratizing access for all institutions to operate and benefit by leveraging the public cloud. It is important to collaborate with our peers to ensure consistency across cloud service providers, ensuring the industry can realize true multi-cloud strategies,”

## Goals of FINOS CCC

> Base on the Common Cloud Controls GitHub repo the project aims to fulfil following goals, 

Reviewer 3: ```suggestion
Based on the Common Cloud Controls GitHub repo the project aims to fulfil following goals: 
```

* Defining Best Practices Around Cloud Security
* One Target For CSPs To Conform To
* Sharing The Burden Of A Common Definition
* A Path Towards Common Implementation
* A Path Towards Certification

> or a more detailed explanation of each project goal and how they are achieved, please refer to the [README.md](https://github.com/finos/common-cloud-controls) file in the project's GitHub repository.

Reviewer 3: ```suggestion
For a more detailed explanation of each project goal and how they are achieved, please refer to the [README.md](https://github.com/finos/common-cloud-controls) file in the project's GitHub repository.
```

> For a more detailed explanation of each project goal and how they are achieved, please refer to the [README.md](https://github.com/finos/common-cloud-controls) file in the project's GitHub repository.

Reviewer 4: I wanted to offer some thoughts on the ordering of the 'What, Why, Goals' sections. I think presenting the goals earlier, right after the 'What,' could create a stronger impact. The goals you’ve outlined are bold and exciting—anyone with a solid understanding of cloud technology knows that achieving these is no small feat, and that's what makes them so compelling. Seeing the goals upfront would immediately spark curiosity about how you plan to achieve them, the challenges you're addressing, and so on.

I also feel like the 'Why' section could be more closely integrated with the goals. Right now, it feels a bit detached—important information, but something I’m tempted to skip. From my perspective, the story you're telling is: 'This is what we’re doing, we started because the financial services sector faces specific challenges around on-prem to cloud migration, and here are the goals we've set to address that.' However, to me this narrative isn't quite flowing.

Bringing the goals forward could help ground the 'Why' section by providing context—why these specific goals? How did they come about? You could explain that these goals were shaped by the unique challenges of the financial services sector, which would then flow naturally into the timeline, where you detail when the project began, when it was open-sourced, etc., before diving deeper into the specifics of what you're actually doing.

I'm _more_ than happy to be challenged on this, as this is merely my perspective and interpretation.

Author: Will reorder. It make sense. Thanks!

Author: Done the reordering. It flows better. Thanks! :)

The project also aims to validate controls through a series of tests. If you are aware of the controls required in your public cloud, you can use the tests provided in the CCC standard to verify whether those controls are properly implemented.

It is envisaged that eventually, CCC will offer certification for CSPs who conform to the standard.

## Controls, Threats and Features

To provide you with a clearer understanding of what a control is, let's take a closer look at a specific example: the control that prevents bucket deletion through irrevocable bucket retention policy in object storage. This control is designed to ensure that critical storage resources, like data buckets, are not accidentally or maliciously deleted, thereby protecting the integrity and availability of stored data. By examining this control, we can better understand what are the associated threats and how to mitigate those risks in cloud environments.

> ~~~ yaml

Reviewer 3: The format has changed, might be worth updating to the latest version?

nist_csf: PR.DS-1
test_requirements:
    tlp_green:
    01: |
        Verify that the bucket cannot be deleted as long as the retention policy is in effect.
    02: |
        Confirm that the bucket retention policy cannot be modified or unset.
~~~

This control defined in the file named `controls.yaml` under object storage [link](https://github.com/finos/common-cloud-controls/blob/main/services/storage/object/controls.yaml). It is mapped to a specific threat within the standard, identified as `CCC.TH06`, which we will explore in more detail later. Additionally, this control is mapped to a NIST control, specified as `PR.DS-1`, which is part of the NIST framework's guidelines for protecting data. There are also specific methods to test whether this control is effectively implemented within your cloud service provider, ensuring that it meets security and compliance standards.

~~~yaml
id: CCC.TH06
title: Accidental or malicious deletion of critical data
description: |
    Critical data can be deleted either through accidental action by authorized users 
    or through malicious intent by threat actors exploiting vulnerabilities or misconfigurations.
features: 
    - CCC.ObjStor.F11

> mitre_attack:

Reviewer 3: The schema has been changed here as well, would be worth updating.
```suggestion
mitre_technique:
```

---
title: Introducing FINOS Common Cloud Controls (CCC)
categories: 
- Open Source
author: smendis-scottlogic

> summary: FINOS Common Cloud Controls (CCC) is an open standard by FINOS, to describe consistent controls for compliance public cloud deployments in financial services sector. The project is sponsored by Scott Logic, aligning with its mission to promote and support open-source initiatives. This is an effort to introduce FINOS CCC and its goals.

Reviewer 2: compliance --> compliant

---

> Since September last year (2023), I have been working on a somewhat unusual software project. The project is sponsored by Scott Logic, aligning with its commitment to open source. I was excited, not only for the opportunity to contribute to an open-source project but also by the idea of playing a role in shaping the cloud itself.

Reviewer 2: Let's try to find a better word than "sponsored" as I think that's a loaded term.

> 

Reviewer 2: project --> project,

> Over the course of a year, we’ve come a long way—transforming what was once a vague, uncertain vision into releasable open standard. This blog post aims to introduce the FINOS Common Cloud Controls (CCC) project, outlining its objectives, timeline, and benefits.

Reviewer 2: Need some spaces either side of the dash

> 

Reviewer 2: releasable --> a releasable

## What is FINOS CCC?

![FINOS_LOGO]({{ site.github.url }}/smendis-scottlogic/assets/finos-logo.png)

As stated on the [official page](https://www.finos.org/common-cloud-controls-project) for the CCC project on the FINOS website, 

> “FINOS Common Cloud Controls (FINOS CCC) is the codename for an open standard project, originally proposed by Citi and now open source under Fintech Open Source Foundation (FINOS), to describe consistent controls for compliant public cloud deployments in the financial services sector”.

> This statement outlines several important aspects of the project. First and foremost, it is designed to cater specific needs of the **financial services sector**, which includes banking, insurance, investment and wealth management, mortgage lending, and more. An **open standard** refers to a set of guidelines or specifications developed collaboratively that can be used freely or with minimal restrictions. **Consistent controls** imply standardized security, compliance, and governance measures applied uniformly across the infrastructure, applications and processes. These controls ensure that policies related to data protection, access management, auditing, encryption, and monitoring are reliably implemented. **Compliance public cloud deployments** involve using public cloud services, such as AWS, Microsoft Azure, or Google Cloud, that meet the regulatory and legal standards required by regulating authorities. 

Reviewer 2: Previous comment (don't know if you've got that...) suggested formatting this as a list - might make it easier to read.

In the past, regulatory requirements for financial institutes mandated that data be stored in highly secured on-premise data centres. However, with the growing adoption of cloud services, the financial services sector is increasingly moving towards adapting the public cloud. Key benefits driving this shift include agility, scalability, cost optimization, accelerated innovation, geographic availability, and enhanced resilience. Some of the potential drawbacks and challenges, particularly given the sensitive nature of the financial data are security concerns, compliance and regulatory challenges, loss of control, downtime, data privacy risks, vendor lock-in and skills gap. 

![PROS_AND_CONS_OF_CLOUD]({{ site.github.url }}/smendis-scottlogic/assets/pros-n-cons.png)

> Let’s elaborate more on few key concerns in moving financial data to the public cloud. Despite the robust security measures cloud providers offer, financial data is highly sensitive, and breaches can be devastating. Public cloud environments are multi-tenant, meaning data from multiple organizations shares the same infrastructure. While cloud providers enforce strict isolation, the risk of data leakage still exists. The cloud provider’s employees may have access to critical data, posing potential insider threats. When it comes to regulatory challenges, regulations in some regions require that sensitive financial data must be stored within national borders. Cloud providers may not offer appropriate data centre locations, or ensure compliance with data residency laws can be complex. Auditing and proving compliance in cloud environments can be more challenging, especially with limited visibility into the provider’s operations.

Reviewer 2: "on a few key"

> According to the Linux Foundation [announcement](https://www.linuxfoundation.org/press/finos-announces-open-standards-project-for-financial-services-common-cloud-controls) on July 27, 2023 Jim Adams, CTO and Head of Technology Infrastructure at Citi, the world’s fifth largest bank, stated

Reviewer 2: I don't get this sentence - what are you trying to say?

Threats in the cloud are reasonably understood. The [MITRE ATT&CK](https://attack.mitre.org/) framework is a globally recognized knowledge base used to understand and analyze the behavior of cyber adversaries. It provides a structured way to describe and categorize the tactics, techniques, and procedures (TTPs) that attackers use to infiltrate and compromise systems. CCC also aims to create a mapping of threats found in Mitre framework with the common features identified by the cloud services taxonomy in their controls. 

> FINOS CCC project uses [OSCAL](https://pages.nist.gov/OSCAL/) (Open Security Controls Assessment Language) developed by NIST (National Institute of Standards and Technology) as their control language. OSCAL utilizes a machine-readable format for defining controls, which facilitates automated assessments, reporting, automated generation of compliance documentation and many more.

Reviewer 2: many --> much

    permissions
control_family: Identity and Access Management
objective: |
    Ensure that uniform bucket-level access is enforced across all 
    object storage buckets. This prevents the use of ad-hoc or 
    inconsistent object-level permissions, ensuring centralized, 
    consistent, and secure access management in accordance with the 
    principle of least privilege.
threats:
    - CCC.TH01 # Access control is misconfigured
    - CCC.ObjStor.TH02 # Improper enforcement of object modification locks
nist_csf: PR.AC-4 # Access permissions and authorizations are managed, incorporating the principles of least privilege and separation of duties
control_mappings:
    CCM:
    - DCS-09 # Access Control
    ISO_27001:
    - 2013 A.9.4.1 # Information Access Restriction
    NIST_800_53:
    - AC-3 # Access Enforcement
    - AC-6 # Least Privilege
test_requirements:
    - id: CCC.ObjStor.C02.TR01
    text: |
        Admin users can configure bucket-level permissions uniformly across 
        all buckets, ensuring that object-level permissions cannot be 
        applied without explicit authorization.
    tlp_levels:
        - tlp_amber
        - tlp_red
~~~

> This control defined in the file [`controls.yaml`](https://github.com/finos/common-cloud-controls/blob/main/services/storage/object/controls.yaml) under object storage. This control is designed to prevent the use of ad-hoc object level permissions in buckets. This control is mapped to specific threats within the standard, identified as `CCC.TH01` and `CCC.ObjStor.TH02`, which we will explore in more detail later. Additionally, this control is mapped to a NIST control, specified as [`PR.AC-4`]( https://csf.tools/reference/nist-cybersecurity-framework/v1-1/pr/pr-ds/pr-ac-4/), which is part of the NIST framework's guidelines for access control. There are also specific methods to test whether this control is effectively implemented within your cloud service provider, ensuring that it meets security and compliance standards.

Reviewer 2: This sentence doesn't appear to go anywhere! Is there a second half that's missing somewhere?

> 

Reviewer 2: ad hoc

~~~yaml
id: CCC.TH01 # Access control is misconfigured
title: Access control is misconfigured
description: |
    An attacker can exploit misconfigured access controls to grant excessive
    privileges or gain unauthorized access to sensitive resources.
features:
    - CCC.F06 # Identity Based Access Control
mitre_technique:
    - T1078 # Valid Accounts
    - T1548 # Abuse Elevation Control Mechanism
    - T1203 # Exploitation for Credential Access
    - T1098 # Account Manipulation
    - T1484 # Domain or Tenant Policy Modification
    - T1546 # Event Triggered Execution
    - T1537 # Transfer Data to Cloud Account
    - T1567 # Exfiltration Over Web Services
    - T1048 # Exfiltration Over Alternative Protocol
    - T1485 # Data Destruction
    - T1565 # Data Manipulation
    - T1027 # Obfuscated Files or Information
~~~

> Let’s examine the threat `CCC.TH01` in the file [`common-threats.yaml`](https://github.com/finos/common-cloud-controls/blob/main/services/common-threats.yaml). This highlights the potential risk where attackers exploit access control to gain high privilege access to data. This is identified as a common threat but applicable to object storage. Hence listed under `common_threats` section in the file [`threats.yaml`](https://github.com/finos/common-cloud-controls/blob/main/services/storage/object/threats.yaml) under object storage. This particular threat is also linked to few specific threat in MITRE ATT&CK framework under the IDs [`T1078`](https://attack.mitre.org/techniques/T1078/), [`T1548`](https://attack.mitre.org/techniques/T1548/), [`T1203`](https://attack.mitre.org/techniques/T1203/), [`T1098`](https://attack.mitre.org/techniques/T1098/), [`T1484`](https://attack.mitre.org/techniques/T1484/), [`T1546`](https://attack.mitre.org/techniques/T1546/), [`T1537`](https://attack.mitre.org/techniques/T1537/), [`T1567`](https://attack.mitre.org/techniques/T1567/), [`T1048`](https://attack.mitre.org/techniques/T1048/), [`T1485`](https://attack.mitre.org/techniques/T1485/), [`T1565`](https://attack.mitre.org/techniques/T1565/), [`T1027`](https://attack.mitre.org/techniques/T1027/), which discus data and access manipulation. This threat is also mapped to a specific feature within the standard identified as `CCC.F06` with the title **Identity Based Access Control**. 

Reviewer 2: Do you mean "linked to a few specific threats"? Or something else? Either way, it needs a little tidying to make the meaning clearer!

~~~yaml
id: CCC.ObjStor.TH02 # Improper enforcement of object modification locks
title: Improper enforcement of object modification locks
description: |
    Attackers may exploit vulnerabilities in object modification locks to
    delete or alter objects despite the lock being in place, leading to data
    loss or tampering.
features:
    - CCC.ObjStor.F09 # Object Modification Locks
mitre_technique:
    - T1027 # Obfuscated Files or Information
    - T1485 # Data Destruction
    - T1490 # Inhibit System Recovery
    - T1491 # Defacement
    - T1565 # Data Manipulation
~~~

> Let's examine the threat `CCC.ObjStor.TH02` in the file named [`threats.yaml`](https://github.com/finos/common-cloud-controls/blob/main/services/storage/object/threats.yaml) under object storage. This is an object storage specific threat that discuss attackers exploiting vulnerabilities in object modification locks to destruct data. This particular threat is also linked to few specific threat in MITRE ATT&CK framework under the IDs [`T1027`](https://attack.mitre.org/techniques/T1027/), [`T1485`](https://attack.mitre.org/techniques/T1485/), [`T1490`](https://attack.mitre.org/techniques/T1490/), [`T1491`](https://attack.mitre.org/techniques/T1491/), [`T1565`](https://attack.mitre.org/techniques/T1565/), which discus data destruction and manipulation. This threat is also mapped to a specific feature within the standard identified as `CCC.ObjStor.F09` with the title **Object Modification Locks**. 

Reviewer 2: Same thing as earlier here - "a few specific threats" perhaps?

~~~yaml
id: CCC.F06 # Identity Based Access Control
title: Identity Based Access Control
description: |
    Provides the ability to determine access to resources based on
    attributes associated with a user identity.
~~~

> The feature `CCC.F06`, found in the file named [`common-features.yaml`](https://github.com/finos/common-cloud-controls/blob/main/services/common-features.yaml) is a common feature that referred by the object storage threat `CCC.TH01` discussed above. This feature is also listed under `common_features` in the file [`features.yaml`](https://github.com/finos/common-cloud-controls/blob/main/services/storage/object/features.yaml) under the object storage, identifying it as a feature in object storage. This functionality of this feature is to control access to the object storage buckets based on identity.

Reviewer 2: referred to by the

~~~yaml
id: CCC.ObjStor.F09 # Object Modification Locks
title: Object Modification Locks
description: |
    Allows locking of objects to disable modification and/or deletion of an
    object for a defined period of time.
~~~

> The feature `CCC.ObjStor.F09`, found in the file named [`features.yaml`](https://github.com/finos/common-cloud-controls/blob/main/services/storage/object/features.yaml) under object storage is an object storage specific feature that referred by the threat `CCC.ObjStor.TH02` discussed above. This functionality of this feature is to provide object locks for data stored in object buckets disabling modifications and/or deletion.

Reviewer 2: referred to

> In summary, if your architecture relies on object storage and its features such as *identity based access storage* and *object modification locks* you are susceptible to threats such as *access control is misconfigured* and *improper enforcement of object modification locks*. To prevent those attacks taking place, it's critical to implement controls such as *enforce uniform bucket-level access to prevent inconsistent* for all your object storage buckets that retain sensitive data, in your financial institute. You can validate whether thess controls are in place by executing validation tests that are listed under the controls.

Reviewer 2: and *object modification locks* --> and *object modification locks*,

> 

Reviewer 2: Don't need that comma

For more details refer to the project's [GitHub](https://github.com/finos/common-cloud-controls) page.

## Scott Logic FINOS CCC Team

> Scott Logic was one of the first organizations to collaborate with the FINOS Foundation in establishing the FINOS CCC, under the leadership of Colin Eberhardt and Robert Griffiths. Rob, as the project’s sponsorship lead at Scott Logic, plays a pivotal role in driving this initiative by being a key member of the FINOS CCC Steering Committee. Alongside Rob, Stevie Shiells, who chairs the Community Structure working group, and I, as the chair of the Taxonomy working group, represent Scott Logic in driving this open-source project. We have received, and continue to receive, valuable contributions over time from our Scott Logic team, including Joshua Isted, Cara Fisher, David Ogle, Mike Smith, Euthyme Ziogas, Daniel Moorhouse, and Ivan Mladjenovic.

Reviewer 2: initiative forward

> 

Reviewer 2: The sentence that starts with "Alongside" feels a little clunky. How about asking an LLM for help to rewrite that one ;-)

---
title: Introducing FINOS Common Cloud Controls (CCC)
categories: 
- Open Source
author: smendis-scottlogic
summary: FINOS Common Cloud Controls (CCC) is an open standard by FINOS, to describe consistent controls for compliant public cloud deployments in the financial services sector. The project is supported by Scott Logic, aligning with its mission to promote and support open-source initiatives. This is an effort to introduce FINOS CCC and its goals.
---

Since September last year (2023), I have been working on a somewhat unusual software project. The project is supported by Scott Logic, aligning with its commitment to open source. I was excited, not only for the opportunity to contribute to an open-source project, but also by the idea of playing a role in shaping the cloud itself.

Over the course of a year, we’ve come a long way — transforming what was once a vague, uncertain vision into a releasable open standard. This blog post aims to introduce the FINOS Common Cloud Controls (CCC) project, outlining its objectives, timeline, and benefits.

## What is FINOS CCC?

![FINOS_LOGO]({{ site.github.url }}/smendis-scottlogic/assets/finos-logo.png)

> As stated on the [official page](https://www.finos.org/common-cloud-controls-project) for the CCC project on the FINOS website, 

Reviewer 5: Be careful with non-specific link text such as "official page" because it doesn't say anything about the link destination, for the purposes of search-engine crawls and hit ranking, or for screen readers. I think this link would be better as "Common Cloud Controls" as that is the title of the site you are linking to.

I also note that this finos site does not have "Common Cloud Controls" or "CCC" as keywords.

To fully understand the above definition, let's explore the key elements highlighted. 

- *Open standard* - set of guidelines or specifications developed collaboratively that can be used freely or with minimal restrictions.
- *Consistent controls* - standardized security, compliance, and governance measures applied uniformly across the infrastructure, applications and processes.
- *Compliant public cloud deployments* - cloud service deployments that adhere to regulatory standards, international laws and industry best practices. Most popular public clouds are AWS, Microsoft Azure, and Google Cloud.
- *Financial services sector* - encompasses a broad range of services related to finance and money management, including banking, mortgages, payment services, investments, insurance and more.

In summary, FINOS CCC project aims to establish a set of guidelines that enforce security, compliance, and governance for public cloud services used by the financial institutions.

## Goals of FINOS CCC

> Based on the Common Cloud Controls [GitHub](https://github.com/finos/common-cloud-controls) repo the project aims to fulfil following goals: 

Reviewer 5: Use "Common Cloud Controls GitHub repo" as link text.

Cloud infrastructure also enhances business agility by allowing companies to quickly adapt to market changes. The pay-as-you-go model offers financial flexibility, where businesses only pay for the resources they actually use, preventing unnecessary expenditures. Cloud providers also typically offer built-in security features and compliance certifications, reducing the burden on internal IT teams to maintain security protocols.

Disaster recovery and data backup solutions are another key benefit, as cloud platforms provide reliable, automated solutions for minimizing downtime in the event of disruptions. These advantages collectively lead to improved business continuity, performance, and scalability in ways that traditional, on-premises setups struggle to match.

## Timeline and Contributors

> **On July 27, 2023, in New York,** FINOS announced the formation of an open standard project, based upon an approach developed by FINOS Platinum Member Citi, to describe consistent controls ([link](https://www.linuxfoundation.org/press/finos-announces-open-standards-project-for-financial-services-common-cloud-controls)). 

Reviewer 5: I suggest you use "FINOS announced the formation of an open standard project to describe common cloud controls" or similar as link text. Definitely do not use "link"!

Reviewer 5: > FINOS announced the formation

This says nothing about what was formed, so will likely only come up in searches for "FINOS announcement". I know that my original suggestion was long, but it contains plenty of keywords. Can possibly shorten it a bit?

> [FINOS announced a new standard for common, consistent cloud controls](https://www.linuxfoundation.org/press/finos-announces-open-standards-project-for-financial-services-common-cloud-controls), based upon an approach developed by FINOS Platinum Member Citi.

> In conjunction with the announcement of formation, **Jim Adams**, *CTO and Head of Technology Infrastructure* at Citi, the world’s fifth largest bank, stated ([link](https://www.linuxfoundation.org/press/finos-announces-open-standards-project-for-financial-services-common-cloud-controls)),

Reviewer 5: This is a tricky one, as it's a quote from another site. At this stage, I'm wondering if a reference list might make sense. Would @pauldykes have an opinion?

FINOS CCC project uses [OSCAL](https://pages.nist.gov/OSCAL/) (Open Security Controls Assessment Language) developed by NIST (National Institute of Standards and Technology) as their control language. OSCAL utilizes a machine-readable format for defining controls, which facilitates automated assessments, reporting, automated generation of compliance documentation and much more.

The project also aims to validate controls through a series of tests. If you are aware of the controls required in your public cloud, you can use the tests provided in the CCC standard to verify whether those controls are properly implemented.

It is envisaged that eventually, CCC will offer certification for CSPs who conform to the standard.

## Controls, Threats and Features

To provide you with a clearer understanding of what a control is, let's take a closer look at a specific example.

~~~ yaml

>  id: CCC.ObjStor.C02 # Enforce uniform bucket-level access to prevent inconsistent permissions

Reviewer 5: There's a leading whitespace here before `id`, I assume not deliberate.

    permissions
control_family: Identity and Access Management
objective: |
    Ensure that uniform bucket-level access is enforced across all 
    object storage buckets. This prevents the use of ad hoc or 
    inconsistent object-level permissions, ensuring centralized, 
    consistent, and secure access management in accordance with the 
    principle of least privilege.
threats:
    - CCC.TH01 # Access control is misconfigured
    - CCC.ObjStor.TH02 # Improper enforcement of object modification locks
nist_csf: PR.AC-4 # Access permissions and authorizations are managed, incorporating the principles of least privilege and separation of duties
control_mappings:
    CCM:
    - DCS-09 # Access Control
    ISO_27001:
    - 2013 A.9.4.1 # Information Access Restriction
    NIST_800_53:
    - AC-3 # Access Enforcement
    - AC-6 # Least Privilege
test_requirements:
    - id: CCC.ObjStor.C02.TR01
    text: |
        Admin users can configure bucket-level permissions uniformly across 
        all buckets, ensuring that object-level permissions cannot be 
        applied without explicit authorization.
    tlp_levels:
        - tlp_amber
        - tlp_red
~~~

> This control defined in the file [`controls.yaml`](https://github.com/finos/common-cloud-controls/blob/main/services/storage/object/controls.yaml) under object storage. This control is designed to prevent the use of ad hoc object level permissions in buckets. This control is mapped to specific threats within the standard, identified as `CCC.TH01` and `CCC.ObjStor.TH02`, which we will explore in more detail later. Additionally, this control is mapped to a NIST control, specified as [`PR.AC-4`]( https://csf.tools/reference/nist-cybersecurity-framework/v1-1/pr/pr-ds/pr-ac-4/), which is part of the NIST framework's guidelines for access control. There are also specific methods to test whether this control is effectively implemented within your cloud service provider, ensuring that it meets security and compliance standards.

Reviewer 5: Typo

> This control _is_ defined

Reviewer 5: Not addressed yet...

> This control defined in the file [`controls.yaml`](https://github.com/finos/common-cloud-controls/blob/main/services/storage/object/controls.yaml) under object storage. This control is designed to prevent the use of ad hoc object level permissions in buckets. This control is mapped to specific threats within the standard, identified as `CCC.TH01` and `CCC.ObjStor.TH02`, which we will explore in more detail later. Additionally, this control is mapped to a NIST control, specified as [`PR.AC-4`]( https://csf.tools/reference/nist-cybersecurity-framework/v1-1/pr/pr-ds/pr-ac-4/), which is part of the NIST framework's guidelines for access control. There are also specific methods to test whether this control is effectively implemented within your cloud service provider, ensuring that it meets security and compliance standards.

Reviewer 5: > Additionally, this control is mapped to [NIST control PR.AC-4](https://csf.tools/reference/nist-cybersecurity-framework/v1-1/pr/pr-ds/pr-ac-4/)

Because PR.AC-4 on its own is pretty meaningless 😅

Reviewer 5: This one?

~~~yaml
id: CCC.TH01 # Access control is misconfigured
title: Access control is misconfigured
description: |
    An attacker can exploit misconfigured access controls to grant excessive
    privileges or gain unauthorized access to sensitive resources.
features:
    - CCC.F06 # Identity Based Access Control
mitre_technique:
    - T1078 # Valid Accounts
    - T1548 # Abuse Elevation Control Mechanism
    - T1203 # Exploitation for Credential Access
    - T1098 # Account Manipulation
    - T1484 # Domain or Tenant Policy Modification
    - T1546 # Event Triggered Execution
    - T1537 # Transfer Data to Cloud Account
    - T1567 # Exfiltration Over Web Services
    - T1048 # Exfiltration Over Alternative Protocol
    - T1485 # Data Destruction
    - T1565 # Data Manipulation
    - T1027 # Obfuscated Files or Information
~~~

> Let’s examine the threat `CCC.TH01` in the file [`common-threats.yaml`](https://github.com/finos/common-cloud-controls/blob/main/services/common-threats.yaml). This highlights the potential risk where attackers exploit access control to gain high privilege access to data. This is identified as a common threat but applicable to object storage. Hence listed under `common_threats` section in the file [`threats.yaml`](https://github.com/finos/common-cloud-controls/blob/main/services/storage/object/threats.yaml) under object storage. This particular threat is also linked to few specific threat in MITRE ATT&CK framework under the IDs [`T1078`](https://attack.mitre.org/techniques/T1078/), [`T1548`](https://attack.mitre.org/techniques/T1548/), [`T1203`](https://attack.mitre.org/techniques/T1203/), [`T1098`](https://attack.mitre.org/techniques/T1098/), [`T1484`](https://attack.mitre.org/techniques/T1484/), [`T1546`](https://attack.mitre.org/techniques/T1546/), [`T1537`](https://attack.mitre.org/techniques/T1537/), [`T1567`](https://attack.mitre.org/techniques/T1567/), [`T1048`](https://attack.mitre.org/techniques/T1048/), [`T1485`](https://attack.mitre.org/techniques/T1485/), [`T1565`](https://attack.mitre.org/techniques/T1565/), [`T1027`](https://attack.mitre.org/techniques/T1027/), which discus data and access manipulation. This threat is also mapped to a specific feature within the standard identified as `CCC.F06` with the title **Identity Based Access Control** which we will explore later. 

Reviewer 5: Typos

> linked to _a_ few specific _threats_ in MITRE"

> which _discuss_ data

Reviewer 5: First one not fixed yet.

~~~yaml
id: CCC.ObjStor.TH02 # Improper enforcement of object modification locks
title: Improper enforcement of object modification locks
description: |
    Attackers may exploit vulnerabilities in object modification locks to
    delete or alter objects despite the lock being in place, leading to data
    loss or tampering.
features:
    - CCC.ObjStor.F09 # Object Modification Locks
mitre_technique:
    - T1027 # Obfuscated Files or Information
    - T1485 # Data Destruction
    - T1490 # Inhibit System Recovery
    - T1491 # Defacement
    - T1565 # Data Manipulation
~~~

> Let's examine the threat `CCC.ObjStor.TH02` in the file [`threats.yaml`](https://github.com/finos/common-cloud-controls/blob/main/services/storage/object/threats.yaml) under object storage. This is an object storage specific threat that discusses attackers exploiting vulnerabilities in object modification locks to destroy data. This particular threat is also linked to a few specific threats in MITRE ATT&CK framework under the IDs [`T1027`](https://attack.mitre.org/techniques/T1027/), [`T1485`](https://attack.mitre.org/techniques/T1485/), [`T1490`](https://attack.mitre.org/techniques/T1490/), [`T1491`](https://attack.mitre.org/techniques/T1491/), [`T1565`](https://attack.mitre.org/techniques/T1565/), which discus data destruction and manipulation. This threat is also mapped to a specific feature within the standard identified as `CCC.ObjStor.F09` with the title **Object Modification Locks** which we will discuss in more detail later. 

Reviewer 5: Typo

> which _discuss_ data destruction

~~~yaml
id: CCC.ObjStor.F09 # Object Modification Locks
title: Object Modification Locks
description: |
    Allows locking of objects to disable modification and/or deletion of an
    object for a defined period of time.
~~~

The feature `CCC.ObjStor.F09`, found in the file [`features.yaml`](https://github.com/finos/common-cloud-controls/blob/main/services/storage/object/features.yaml) under object storage is an object storage specific feature that is referred to by the threat `CCC.ObjStor.TH02` discussed above. The functionality of this feature is to provide object locks for data stored in object buckets disabling modifications and/or deletion.

> In summary, if your architecture relies on object storage and its features such as *identity based access control* and *object modification locks*, you are susceptible to threats such as *misconfigured access control* and *improper enforcement of object modification locks*. To prevent those attacks from taking place, it's critical to implement controls such as *enforce uniform bucket-level access to prevent inconsistent* for all your object storage buckets that retain sensitive data in your financial institute. You can validate whether these controls are in place by executing validation tests that are listed under the controls.

Reviewer 5: > enforce uniform bucket-level access to prevent inconsistent ...

Some word missing here? Inconsistent what?

Author: Yes _permission_ was missing after _inconsistent_

> For more details refer to the project's [GitHub](https://github.com/finos/common-cloud-controls) page.

Reviewer 5: As before, need informative link text here.

## Scott Logic FINOS CCC Team

> Scott Logic was among the first organizations to collaborate with the FINOS Foundation in establishing the FINOS CCC, under the leadership of Colin Eberhardt and Robert Griffiths. As the project's sponsorship lead at Scott Logic, Rob plays a key role in advancing this initiative as a member of the FINOS CCC Steering Committee. Alongside Rob, Stevie Shiells, chair of the Community Structure working group, and I, as chair of the Taxonomy working group, represent Scott Logic in leading this open-source project. We've also received, and continue to benefit from, valuable contributions from our Scott Logic team members: Joshua Isted, Cara Fisher, David Ogle, Mike Smith, Euthyme Ziogas, Daniel Moorhouse, and Ivan Mladjenovic.

Reviewer 5: Hmm. Is it possible / trivial to link each of these collaborators to their own blog posts (those that have)? Seems a shame to miss that opportunity.

Reviewer 5: Brilliant!
